core/mmu.c

```diff
@@ -382,15 +382,10 @@ int create_mem_mapping(struct mm_struct *mm, vir_addr_t addr,
 		phy_addr_t phy, size_t size, unsigned long flags)
```

Comment:
create_mem_mapping is exported to other module, it can be called directly. I think spin_lock()/spin_unlock shoud be in  create_mem_mapping.

Reply:
Currently, it is only used in `create_host_mapping` and `create_guest_mapping`. I moved the lock outside. Because `iommu_iotlb_flush_all` also needs to hold the lock of `mm`.

Reply:
@lemin9538 Currently, it is only used in `create_host_mapping` and `create_guest_mapping`. I moved the lock outside. Because `iommu_iotlb_flush_all` also needs to hold the lock of `mm`.

Reply:
1 - can iommu_iotlb_flush_all() put into destroy_guest_mapping()? minos will flush tlb when umap the memory.

2 - if need to move the lock outside, please also change create_mem_mapping()/destroy_mem_mapping() as static function.

Reply:
@lemin9538 You are right, what about the following?

1. Hold the lock of `mm` in `iommu_iotlb_flush_all`
2. Call `iommu_iotlb_flush_all` in `destroy_guest_mapping`

Is there any other place where the page table will be modified?

Reply:
> @lemin9538 You are right, what about the following?
> 
> 1. Hold the lock of `mm` in `iommu_iotlb_flush_all`

what this mean? it means in iommu_iotlb_flush_all() spin_lock/spin_unlock will be called ?

> 2. Call `iommu_iotlb_flush_all` in `destroy_guest_mapping`
ok for me.

> 
> Is there any other place where the page table will be modified?
currently I remember all via create_host/guest_mapping() or destroy_host/guest_mapping().

Reply:
@lemin9538 I will make some changes, you can check back later.